src/uuid.ts

```diff
@@ -1,57 +1,219 @@
+   */
+  static generate(): Buffer {
+    const buffer = Buffer.alloc(BYTE_LENGTH);
+    uuidV4Generate(null, buffer, 0);
+
+    return buffer;
+  }
+
+  /**
+   * Creates an UUID from a hex string representation of an UUID.
+   * @param hexString - 36 character hex string (dashes included, format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
+   */
+  static createFromHexString(hexString: string): UUID {
+    const buffer = uuidHexStringToBuffer(hexString);
+    return new UUID(buffer);
   }
 
-  let v;
-  const arr = new Uint8Array(16);
+  /** Returns the UUID id as a 36 character hex string representation, including dashes */
+  toHexString(): string {
+    if (UUID.cacheHexString && this.__id) {
+      return this.__id;
+    }
 
-  // Parse ########-....-....-....-............
-  arr[0] = (v = parseInt(uuid.slice(0, 8), 16)) >>> 24;
-  arr[1] = (v >>> 16) & 0xff;
-  arr[2] = (v >>> 8) & 0xff;
-  arr[3] = v & 0xff;
+    const uuidHexString = [
+      this.id.toString('hex', 0, 4),
+      this.id.toString('hex', 4, 6),
+      this.id.toString('hex', 6, 8),
+      this.id.toString('hex', 8, 10),
+      this.id.toString('hex', 10, 16)
+    ].join('-');
 
-  // Parse ........-####-....-....-............
-  arr[4] = (v = parseInt(uuid.slice(9, 13), 16)) >>> 8;
-  arr[5] = v & 0xff;
+    if (UUID.cacheHexString) {
+      this.__id = uuidHexString;
+    }
 
-  // Parse ........-....-####-....-............
-  arr[6] = (v = parseInt(uuid.slice(14, 18), 16)) >>> 8;
-  arr[7] = v & 0xff;
+    return uuidHexString;
+  }
 
-  // Parse ........-....-....-####-............
-  arr[8] = (v = parseInt(uuid.slice(19, 23), 16)) >>> 8;
-  arr[9] = v & 0xff;
+  /**
+   * Converts the id into a 36 character (dashes included) hex string, unless a encoding is specified.
+   * @internal
+   */
+  toString(encoding?: string): string {
+    return encoding ? this.id.toString(encoding) : this.toHexString();
+  }
+
+  /**
+   * Converts the id into its JSON string representation. A 36 character (dashes included) hex string in the format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+   * @internal
+   */
+  toJSON(): string {
+    return this.toHexString();
+  }
 
-  // Parse ........-....-....-....-############
-  // (Use "/" to avoid 32-bit truncation when bit-shifting high-order bytes)
-  arr[10] = ((v = parseInt(uuid.slice(24, 36), 16)) / 0x10000000000) & 0xff;
-  arr[11] = (v / 0x100000000) & 0xff;
-  arr[12] = (v >>> 24) & 0xff;
-  arr[13] = (v >>> 16) & 0xff;
-  arr[14] = (v >>> 8) & 0xff;
-  arr[15] = v & 0xff;
+  /**
+   * Compares the equality of this UUID with `otherID`.
+   *
+   * @param otherId - UUID instance to compare against.
+   */
+  equals(otherId: string | Buffer | UUID): boolean {
+    if (!otherId) {
+      return false;
+    }
 
-  return arr;
+    if (otherId instanceof UUID) {
+      return otherId.id.equals(this.id);
+    }
+
+    try {
+      return new UUID(otherId).id.equals(this.id);
+    } catch (_) {
```

Comment:
I think you can do with just `catch { }` here and the other places

Reply:
I'm so used to linters screaming at me for that... 🤷‍♂️